packages/mattermost-redux/src/actions/users.ts

```diff
@@ -113,57 +114,60 @@ export function loadMe(): ActionFunc {
         const serverVersion = state.entities.general.serverVersion || Client4.getServerVersion();
         dispatch(setServerVersion(serverVersion));
 
-        let responseData: CurrentUserInfoQueryResponseType['data'] | null = null;
+        let userId: UserProfile['id'] = '';
+        let userRolesAsString = '';
+
         try {
-            const {data} = await Client4.fetchWithGraphQL<CurrentUserInfoQueryResponseType>(currentUserInfoQuery);
-            responseData = data;
-        } catch (error) {
-            dispatch(logError(error));
-            return {error};
-        }
+            const {data, errors} = await Client4.fetchWithGraphQL<CurrentUserInfoQueryResponseType>(currentUserInfoQuery);
 
-        if (!responseData) {
-            return {data: false};
-        }
+            if (errors || !data) {
+                throw new Error('Error returned in fetching current user info with graphql');
+            }
 
-        dispatch(
-            batchActions([
-                {
-                    type: GeneralTypes.CLIENT_LICENSE_RECEIVED,
-                    data: responseData.license,
-                },
-                {
-                    type: GeneralTypes.CLIENT_CONFIG_RECEIVED,
-                    data: responseData.config,
-                },
-                {
-                    type: UserTypes.RECEIVED_ME,
-                    data: transformToRecievedMeReducerPayload(responseData.user),
-                },
-                {
-                    type: RoleTypes.RECEIVED_ROLES,
-                    data: transformToRecievedRolesReducerPayload(responseData.user.roles, responseData.teamMembers),
-                },
-                {
-                    type: PreferenceTypes.RECEIVED_ALL_PREFERENCES,
-                    data: responseData.user.preferences,
-                },
-                {
-                    type: TeamTypes.RECEIVED_TEAMS_LIST,
-                    data: transformToRecievedTeamsListReducerPayload(responseData.teamMembers),
-                },
-                {
-                    type: TeamTypes.RECEIVED_MY_TEAM_MEMBERS,
-                    data: transformToRecievedMyTeamMembersReducerPayload(responseData.teamMembers, responseData.user.id),
-                },
-            ]),
-        );
+            dispatch(
```

Comment:
Could you actually move dispatch and everything below it out of the try/catch? It's usually a good idea to minimize the amount of stuff that's inside a try/catch since it can accidentally silence errors you might not have expected to catch. For example, if the `transformToRecievedMeReducerPayload` threw an error, it'd probably look like a network error in the logs because that's in here

Reply:
yes agreed. I have moved the dispatch out of try/catch. But kept the transform function there since i didnt wanted to copy the object with reference, `responseData = data`.

Reply:
What's the reason that you wanted to get rid of the separate `responseData`? 0/5 either way, but I'm curious of the reason for that since defining all those extra variables seems like it was more work for you

Reply:
But to copy the `responseData` since its inside the try/catch, out side of it. we copy the object with reference right?